### DISCUSSION TITLE
Allow sharing recipes with other apps in android (conform to either Schema.org or JSON-LD)

### DISCUSSION THREAD

**Discussion by franckleveque** [mealie-recipes/mealie#4728] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I would be great to be able to share a recipe with other apps on Android like get bring that supports Schema.org or JSON-LD format.

We generally use getbring.com for our shopping list and currently we must transfer manually all ingredients one by one. 

Getbring.com offer the possibility to import a recipe that conform to either Schema.org or JSON-LD in order to extract the ingredients to buy however the recipes from mealie are not understood by the app.

### Please provide a concise description of the feature that would resolve your issue.

Provide a recipe page that conform to either Schema.org or JSON-LD in order to allow sharing with other app that supports the format.

### Please consider and list out some caveats or tradeoffs made in your design decision

I don't see any problem with adding the shema.org tags to allow parsing the recipe by external apps

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Mealie already supports being scraped, as it exposes the schema.org recipe data. We also support Bring! integration from Mealie: https://docs.mealie.io/documentation/getting-started/features/#recipe-actions

**Comment by franckleveque** [comment]
Hi thanks for the answer, just checked the documentation it seems to be
what I need but I don't see how to enable it on mealie.

Is it a new feature of 2.4.0 ?

Is there a documentation to know how to enable the feature?

Thanks for your help.

Le dim. 15 déc. 2024, 19:02, Michael Genson ***@***.***> a
écrit :

**Reply by michael-genson** [reply]
It's been around since before v2, you add integrations in the data management page (see the section on the data management page in the linked docs)

**Comment by franckleveque** [comment]
Hi,

Again thanks for your answer, I was able to find how to set up the recipe
actions.

I finally found the doc in MR
https://github.com/mealie-recipes/mealie/pull/3448

The url given there to link to bring is
https://api.getbring.com/rest/bringrecipes/deeplink?url=${url}&source=web

The good thing is that it correctly open my bring app however, I don't have
the ingredients imported. When I check the messages of the app it states
the imported recipe has no valid format.

I run a local instance of mealie. If I understood correctly the comments in
the MR it seems that only instance of mealie that opened to the web can
share the recipe.

Did I understood correctly or am I missing some information?

Thanks

Le dim. 15 déc. 2024, 19:20, Michael Genson ***@***.***> a
écrit :

**Comment by michael-genson** [comment]
Currently your site has to be publicly accessible for the integration to work. There's [an open PR to fix this](https://github.com/mealie-recipes/mealie/pull/4676)

**Comment by franckleveque** [comment]
Hi,

Thank you for your answer! I checked the pull request you referenced, but
it seems to address the public/private functionalities of groups and
families.

In my specific case, all elements are public, but the site itself is only
accessible on my local network. I don’t want to expose external access
through my router’s firewall, so I'm not sure this PR will solve my issue.

That said, I did come across this repository: felixschndr/mealie-bring-api:
A simple webserver that allows ingredients sent by mealie to be added to a
bring shopping list <https://github.com/felixschndr/mealie-bring-api> from
this opened discussion Bring API integration that does not require the
Mealie instance to be publicly available · mealie-recipes/mealie ·
Discussion #3627 <https://github.com/mealie-recipes/mealie/discussions/3627>.
It provides an intermediary server to connect to the Bring REST API and
enables usage within a local network.

I wanted to share this in case it’s relevant or could inspire a similar
solution.

Regards

Le dim. 15 déc. 2024 à 20:33, Michael Genson ***@***.***> a
écrit :
